pkg/docker/container.go

```diff
@@ -35,7 +35,19 @@ func CheckContainer(ctx context.Context, ow *rpc.OutputWriter, cli *client.Clien
 		return nil, err
 	}
 
-	c := containers[0]
+	var c *types.Container
+	for i, cont := range containers {
+		for _, contname := range cont.Names {
+			if contname == name || contname == "/"+name {
```

Comment:
We have the `/`, because:

```
Apr  1 11:24:52.311521  DEBUG   container found {"req_id": "6eb02cc8", "runner": "local:docker", "container_name": "prometheus-pushgateway", "container_id": "4ea2c9c4365173dbf6143ea96d86ac7e0ffba3d069fe1eb118bdf56ffcb9603d", "state": "run
ning"}
Apr  1 11:24:52.313064  DEBUG   checking state of container     {"req_id": "6eb02cc8", "runner": "local:docker", "container_name": "testground-redis"}
([]types.Container) (len=1 cap=4) {
 (types.Container) {
  ID: (string) (len=64) "5ee41f7b916b36d6d863888136340dd3fffadc8f947f73a6da0e91d703f606d8",
  Names: ([]string) (len=1 cap=4) {
   (string) (len=26) "/testground-redis-exporter"
  },
  Image: (string) (len=22) "bitnami/redis-exporter",
  ImageID: (string) (len=71) "sha256:a3a387238d9cb971a7de2e89de66170443d9464ecad24397866bbe4152c281c1",
  Command: (string) (len=57) "redis_exporter --redis.addr redis://testground-redis:6379",
  Created: (int64) 1585140672,
  Ports: ([]types.Port) (len=1 cap=4) {
   (types.Port) {
    IP: (string) (len=7) "0.0.0.0",
    PrivatePort: (uint16) 9121,
    PublicPort: (uint16) 32771,
    Type: (string) (len=3) "tcp"
   }
  },
  SizeRw: (int64) 0,
  SizeRootFs: (int64) 0,
  Labels: (map[string]string) (len=1) {
   (string) (len=10) "maintainer": (string) (len=32) "Bitnami <containers@bitnami.com>"
  },
  State: (string) (len=7) "running",
  Status: (string) (len=13) "Up 11 minutes",
  HostConfig: (struct { NetworkMode string "json:\",omitempty\"" }) {
   NetworkMode: (string) (len=64) "4e4c4210774f35a0ac9e081f93c2c4543ef0e587cce6d889ac418dc7d67561a6"
  },
  NetworkSettings: (*types.SummaryNetworkSettings)(0xc0006260b0)({
   Networks: (map[string]*network.EndpointSettings) (len=1) {
    (string) (len=18) "testground-control": (*network.EndpointSettings)(0xc0006bc000)({
     IPAMConfig: (*network.EndpointIPAMConfig)(<nil>),
     Links: ([]string) <nil>,
     Aliases: ([]string) <nil>,
     NetworkID: (string) (len=64) "4e4c4210774f35a0ac9e081f93c2c4543ef0e587cce6d889ac418dc7d67561a6",
     EndpointID: (string) (len=64) "65c424ac1de52a268bc926f7f53ae67af23ee00ca838acc8057c8867ff929c89",
     Gateway: (string) (len=10) "192.18.0.1",
     IPAddress: (string) (len=10) "192.18.0.5",
     IPPrefixLen: (int) 16,
     IPv6Gateway: (string) "",
     GlobalIPv6Address: (string) "",
     GlobalIPv6PrefixLen: (int) 0,
     MacAddress: (string) (len=17) "02:42:c0:12:00:05",
     DriverOpts: (map[string]string) <nil>
    })
   }
  }),
  Mounts: ([]types.MountPoint) {
  }
 }
}
```

I am not sure why we have a `[]string` for `container.Names`...

Reply:
on the commandline, you can filter with regular expressions, so rather than searching for `testground-redis` and getting  a response for for redis and redis-exporter you can filter for `^testground-redis$` and get a single response back. 

This might also work rather than generating a regex
https://godoc.org/github.com/docker/engine-api/types/filters#Args.ExactMatch

Reply:
if we can do `exact match`, that'd be perfect. it must be possible, so I agree that this PR is rather crap, and we should scratch it, and use to proper filter arg.

Reply:
(for context: i just coded this quickly so I can work on the benchmarks testplan, and figured i might open it as a draft PR)